### DISCUSSION TITLE
Question(mux-player-react): Setting default subtitles on initial load?

### DISCUSSION THREAD

**Discussion by vojtaholik** [muxinc/elements#426] [Q&A]
We're [storing user preference for `playbackRate`](https://github.com/skillrecordings/products/blob/main/apps/total-typescript/src/hooks/use-mux-player.tsx#L127-L133) and I was wondering if you have an example of achieving something similar with subtitles. I could check for `TextTrack` events on video element, but it's pretty cumbersome since they're being loaded async.

**Comment by cjpillsbury** [comment]
Hey @vojtaholik sorry for the slow response! Would `defaultHiddenCaptions` do the trick?

> Hide captions by default instead of showing them on initial load (when available)

From: https://github.com/muxinc/elements/blob/main/packages/mux-player-react/REFERENCE.md#props

**Reply by dylanjha** [reply]
@vojtaholik is your user's preference around showing vs. not showing the subtitles? Or were you trying to save a preference around *which* subtitles track gets shown?

**Reply by vojtaholik** [reply]
@dylanjha being able to tell whether a user turned subtitles on or off would be good enough for us since we only provide a single (english) subtitles at the moment. so it's just a toggle. improving this in the future so that it supports multiple subtitles would for sure be very nice.

@cjpillsbury I think that would do the trick _(for videos with a single subtitles option)_, **the problem is I cannot easily track whether they've enabled them or not in the player's UI.** or am I missing something? I'd love an exposed `onTextTrackChange` handler.

thanks!

**Reply by gkatsev** [reply]
mux-player does expose `textTracks` now, which is currently the native TextTrackList object. Though, since you're using mux-player-react, you'll need to get it via the mux-player ref. You could then listen to the `change` event to know when a track was turned on/off.
